spec/commands/v2/put_content_spec.rb

```diff
@@ -59,5 +58,83 @@
       result = described_class.call(payload)
       expect(result.data[:version]).to eq(1)
     end
+
+    context "when there's an existing gone on the path already" do
+      before do
+        create(:gone_draft_content_item, base_path: base_path)
+      end
+
+      it "replaces the existing content" do
+        described_class.call(payload)
+
+        stored_content_item = DraftContentItem.find_by(base_path: base_path)
+        expect(stored_content_item.content_id).to eq(content_id)
+      end
+    end
+
+    context "when there's an existing redirect on the path already" do
+      before do
+        create(:redirect_draft_content_item, base_path: base_path)
+      end
+
+      it "replaces the existing content" do
+        described_class.call(payload)
+
+        stored_content_item = DraftContentItem.find_by(base_path: base_path)
+        expect(stored_content_item.content_id).to eq(content_id)
+      end
+    end
+
+    context "when a gone item wants to replace a content item" do
+      before do
+        create(:draft_content_item, base_path: base_path)
+      end
+
+      let(:payload) {
+        FactoryGirl.build(:gone_draft_content_item,
+          content_id: content_id,
+          base_path: base_path
+        ).as_json.deep_symbolize_keys
+      }
+
+      it "replaces the existing content" do
+        described_class.call(payload)
+
+        stored_content_item = DraftContentItem.find_by(base_path: base_path)
+        expect(stored_content_item.content_id).to eq(content_id)
+      end
+    end
+
+    context "when a redirect item wants to replace a content item" do
+      before do
+        create(:draft_content_item, base_path: base_path)
+      end
+
+      let(:payload) {
+        FactoryGirl.build(:redirect_draft_content_item,
+          content_id: content_id,
+          base_path: base_path
+        ).as_json.deep_symbolize_keys
+      }
+
+      it "replaces the existing content" do
+        described_class.call(payload)
+
+        stored_content_item = DraftContentItem.find_by(base_path: base_path)
+        expect(stored_content_item.content_id).to eq(content_id)
+      end
+    end
+
+    context "when a content item wants to replace a content item" do
+      before do
+        create(:draft_content_item, base_path: base_path)
+      end
+
+      it "does not replace the existing content" do
+        expect {
+          described_class.call(payload)
+        }.to raise_error(CommandRetryableError)
```

Comment:
Not part of this change necessarily, but we do probably want to return a different kind of error here; there's no point in retrying this call, it will never succeed.